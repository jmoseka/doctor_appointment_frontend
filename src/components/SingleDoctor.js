import PropTypes from 'prop-types';
import insta from '../images2/insta-icon-home.png';
import fb from '../images2/fb-icon-home.png';
import twitter from '../images2/twitter-icon-home.png';

const social = [
  { icon: insta },
  { icon: fb },
  { icon: twitter },
];

const SingleDoctor = ({
  name, image, location, description,
}) => (
  <div className="doctors1">
    <div className="image-div">
      <img src={image} alt="Profile of doctor" className="doctors-images" />
    </div>
    <div className="doctor-data">
      <h4>{name}</h4>
      <p>.........................</p>
      <p>
        {description}
      </p>
      <p>{location}</p>
      <div className="social-net">
        {social.map((item) => (
          <a href="/" key={item.icon}>
            <img src={item.icon} alt="social" className="w-8" />
          </a>
        ))}
      </div>
    </div>
  </div>
);

SingleDoctor.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default SingleDoctor;
