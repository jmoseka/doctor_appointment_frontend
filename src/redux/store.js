import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import { SignUpReducer } from './auth/register';
import { LoginUserReducer } from './auth/login';
import myAppointmentReducer from './my_appointment_redux/my_appointment';

const reducer = combineReducers({
  user: SignUpReducer,
  login: LoginUserReducer,
  my_appointment: myAppointmentReducer,
});

const store = createStore(
  reducer, applyMiddleware(logger, thunk),
);

export default store;
