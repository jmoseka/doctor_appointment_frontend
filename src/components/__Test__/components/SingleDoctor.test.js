import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../redux/store';
import SingleDoctor from '../../SingleDoctor';

describe('SingleDoctor', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <SingleDoctor />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
