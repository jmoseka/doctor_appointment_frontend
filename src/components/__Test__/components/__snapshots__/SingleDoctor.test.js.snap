// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleDoctor renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="doctors1"
      >
        <div
          class="image-div"
        >
          <img
            alt="Profile of doctor"
            class="doctors-images"
          />
        </div>
        <div
          class="doctor-data"
        >
          <h4 />
          <p>
            .........................
          </p>
          <p />
          <p />
          <div
            class="social-net"
          >
            <a
              href="/"
            >
              <img
                alt="social"
                class="w-8"
                src="insta-icon-home.png"
              />
            </a>
            <a
              href="/"
            >
              <img
                alt="social"
                class="w-8"
                src="fb-icon-home.png"
              />
            </a>
            <a
              href="/"
            >
              <img
                alt="social"
                class="w-8"
                src="twitter-icon-home.png"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="doctors1"
    >
      <div
        class="image-div"
      >
        <img
          alt="Profile of doctor"
          class="doctors-images"
        />
      </div>
      <div
        class="doctor-data"
      >
        <h4 />
        <p>
          .........................
        </p>
        <p />
        <p />
        <div
          class="social-net"
        >
          <a
            href="/"
          >
            <img
              alt="social"
              class="w-8"
              src="insta-icon-home.png"
            />
          </a>
          <a
            href="/"
          >
            <img
              alt="social"
              class="w-8"
              src="fb-icon-home.png"
            />
          </a>
          <a
            href="/"
          >
            <img
              alt="social"
              class="w-8"
              src="twitter-icon-home.png"
            />
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
